clambcc/clambc-compiler.py

```diff
@@ -491,109 +526,144 @@ def createInputSourceFile(clangLLVM: ClangLLVM, name: str, args: list, options:
     return res
 
 
+INTERNALIZE_API_LIST=[ "_Z10entrypointv"
+        , "entrypoint"
+        , "__clambc_kind"
+        , "__clambc_virusname_prefix"
+        , "__clambc_virusnames"
+        , "__clambc_filesize"
+        , "__clambc_match_counts"
+        , "__clambc_match_offsets"
+        , "__clambc_pedata"
+        , "__Copyright" 
+        ]
+
+OPTIMIZE_OPTIONS = ["-S"
+        , "--disable-loop-unrolling"
+        , " --disable-i2p-p2i-opt"
+        , " --disable-loop-unrolling"
+        , " --disable-promote-alloca-to-lds"
+        , " --disable-promote-alloca-to-vector"
+        , " --disable-simplify-libcalls"
+        , " --disable-tail-calls"
+        , " --vectorize-slp=false"
+        , " --vectorize-loops=false"
+        , " -internalize-public-api-list=\"%s\"" % ','.join(INTERNALIZE_API_LIST)
+        ]
+
+#TODO: Remove this when we properly handle opaque pointers.
+OPTIMIZE_OPTIONS.append("-opaque-pointers=0")
+
+OPTIMIZE_PASSES = ["function(mem2reg)"
+        , 'verify'
+#        , 'clambc-remove-undefs' #TODO: MAY NOT BE NEEDED 
+#        , 'verify'
+        , 'clambc-preserve-abis'
+        , 'verify'
+        , 'default<O3>'
+        , 'globalopt'
+        , 'clambc-preserve-abis' #remove fake function calls because O3 has already run
+        , 'verify'
+        , 'clambc-remove-unsupported-icmp-intrinsics'
+        , 'verify'
+        , 'clambc-remove-usub'
+        , 'verify'
+        , 'clambc-remove-fshl'
+        , 'verify'
+        , 'clambc-lowering-notfinal' # perform lowering pass
+        , 'verify'
+        , 'lowerswitch'
+        , 'verify'
+        , 'clambc-remove-icmp-sle'
+        , 'verify'
+        , 'function(clambc-verifier)'
+        , 'verify'
+        , 'clambc-remove-freeze-insts'
+        , 'verify'
+        , 'clambc-lowering-notfinal'  # perform lowering pass
+        , 'verify'
+        , 'clambc-lcompiler-helper' #compile the logical_trigger function to a
+        , 'verify'
+        , 'clambc-lcompiler' #compile the logical_trigger function to a
+        , 'verify'
+        , 'internalize'
+        , 'verify'
+        , 'clambc-rebuild'
+        , 'verify'
+        , 'clambc-trace'
+        , 'verify'
+        , 'clambc-outline-endianness-calls'
+        , 'verify'
+#        , 'clambc-change-malloc-arg-size' #TODO: MAY NOT BE NEEDED 
```

Comment:
Can you elaborate in the comment why it may not be needed? 

Or have a follow-up commit that removes it and explains why in the commit message?  Then if it turns out we need it again, it can be resurrected with a `git revert`.

Reply:
done